### PR TITLE
Leverage the WordPress e2e testing framework to run end-to-end tests against the plugin

### PR DESCRIPTION
**Pull request by jblz**
## Description

Simulate actual user actions as they interact with this plugin. Tests are run against a "real" WordPress instance and activities are performed in a "real" browser. 

## Motivation and Context

The idea is that we can provide confidence that changes going forward have the intended effect on the DOM and rendered content that plugin users and site visitors will see under various specific conditions.

## How Has This Been Tested?

See the [How to Run](https://github.com/Parsely/wp-parsely/blob/try-e2e-testing/tests/e2e/README.md#how-to-run) section of the README in this branch.

This is also hooked into a Github action to run on pushes and PRs.

## Screenshots (if appropriate):

https://user-images.githubusercontent.com/1587282/132934198-cc24ed30-bb10-4c8c-8623-6cc3287250f9.mp4

## Types of changes

New developer-specific feature (non-breaking change which adds functionality)

### CODE REVIEW

File: tests/e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## How it Works
+
+In order for the tests to do their job, they need a back end that simulates a WordPress instance. To that end, we spin up a bare-bones containerized site and configure it to the default values for the WordPress `e2e-tests` helper.
+
+Once there's a functioning back end, we leverage the `@wordpress/scripts` utility's [built-in functionality](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#test-e2e) to launch a browser via [puppeteer](https://pptr.dev/).
+
+The tests use the using the [jest framework](https://jestjs.io/) to drive a user flow and assert on expected outcomes. In addition to the [puppeteer API](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md), there are a number of helpers to accomplish frequently performed tasks in the [`@wordpress/e2e-test-utils` package](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-e2e-test-utils/).
```
Comment (GaryJones) [inline tests/e2e/README.md]:
```suggestion
The tests use the [Jest framework](https://jestjs.io/) to drive a user flow and assert on expected outcomes. In addition to the [Puppeteer API](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md), there are a number of helpers to accomplish frequently performed tasks in the [`@wordpress/e2e-test-utils` package](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-e2e-test-utils/).
```

File: .eslintrc [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
 	"env": { "browser": true },
-	"extends": [ "plugin:@wordpress/eslint-plugin/recommended-with-formatting" ]
+	"extends": [ "plugin:@wordpress/eslint-plugin/recommended-with-formatting" ],
+	"settings": {
+		"jest": { "version": "latest" }
+	}
```
Comment (jblz) [inline .eslintrc:6]: TODO: Perhaps pull this out to another PR.  My IDE was not properly running eslint without this change for some reason.

Comment (GaryJones) [reply .eslintrc:6]: I'm fine with this in this PR.

File: tests/e2e/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# End-to-end (e2e) Tests for the wp-parsely Plugin
+
+This suite is meant to simulate actual user actions as they interact with this plugin. Tests are run against a "real" WordPress instance and activities are performed in a "real" browser. The idea is that we can provide confidence that changes going forward have the intended effect on the DOM and rendered content that plugin users and site visitors will see under various specific conditions.
+
+## How it Works
+
+In order for the tests to do their job, they need a back end that simulates a WordPress instance. To that end, we spin up a bare-bones containerized site and configure it to the default values for the WordPress `e2e-tests` helper.
+
+Once there's a functioning back end, we leverage the `@wordpress/scripts` utility's [built-in functionality](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#test-e2e) to launch a browser via [puppeteer](https://pptr.dev/).
```
Comment (GaryJones) [inline tests/e2e/README.md]:
```suggestion
Once there's a functioning back end, we leverage the `@wordpress/scripts` utility's [built-in functionality](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#test-e2e) to launch a browser via [Puppeteer](https://pptr.dev/).
```

File: tests/e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+## How to Run
+
+- Provision the back end
+
+  - Make sure [`docker-compose` is installed](https://docs.docker.com/compose/install/).
+  - From the top-level directory, run:
+
+    `docker-compose -f tests/e2e/docker-compose.yml up`
+
+    Tip: if you cd into this directory, you can simply run `docker-compose` & omit the `-f tests/e2e/docker-compose.yml` from these commands
```
Comment (GaryJones) [inline tests/e2e/README.md]:
```suggestion
    Tip: if you `cd` into this directory, you can simply run `docker-compose` and omit the `-f tests/e2e/docker-compose.yml` from these commands
```

File: .github/workflows/e2e-tests.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  test:
+    name: e2e against WordPress latest
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Spin up WordPress
+        run: docker-compose -f tests/e2e/docker-compose.yml up &
```
Comment (GaryJones) [inline .github/workflows/e2e-tests.yml:23]: What does the trailing `&` do here?
(It's not included in the readme for manual running the tests.)

File: tests/e2e/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  - Once you see a line that says:
+
+    `Success: WordPress installed successfully.`
+
+    ..and a tailing HTTP server log, you may proceed.
```
Comment (GaryJones) [inline tests/e2e/README.md]:
```suggestion
    ...and a tailing HTTP server log, you may proceed.
```

File: tests/e2e/README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  - Make sure [`docker-compose` is installed](https://docs.docker.com/compose/install/).
+  - From the top-level directory, run:
+
+    `docker-compose -f tests/e2e/docker-compose.yml up`
+
+    Tip: if you cd into this directory, you can simply run `docker-compose` & omit the `-f tests/e2e/docker-compose.yml` from these commands
+
+  - Once you see a line that says:
+
+    `Success: WordPress installed successfully.`
```
Comment (GaryJones) [inline tests/e2e/README.md:28]: Is this a different message to the "WordPress is ready!" message in the GitHub workflow?

**Review by GaryJones — COMMENTED** [review]
Minor suggestions, but I love where this is heading! Very exciting.

File: .github/workflows/e2e-tests.yml (hunk 5 above)
Comment (pauarge) [reply .github/workflows/e2e-tests.yml:23]: I've tested the removal of it and it breaks the GitHub Actions workflow.

**Review by pauarge — APPROVED** [review]
LGTM
